test/e2e_node/topology_manager_test.go

```diff
@@ -350,27 +499,231 @@ func runTopologyManagerSuiteTests(f *framework.Framework) {
+				deviceLimit:   deviceAmount,
+			},
+		}
+
+		podName := fmt.Sprintf("gu-pod-%d", podID)
+		framework.Logf("creating pod %s attrs %v", podName, ctnAttrs)
+		pod := makeTopologyManagerTestPod(podName, numalignCmd, ctnAttrs)
+		pod = f.PodClient().CreateSync(pod)
+		framework.Logf("created pod %s", podName)
+		pods = append(pods, pod)
+	}
+
+	for podID := 0; podID < numPods; podID++ {
+		validatePodAlignment(f, pods[podID], numaNodes)
+	}
+
+	for podID := 0; podID < numPods; podID++ {
+		pod := pods[podID]
+		framework.Logf("deleting the pod %s/%s and waiting for container %s removal",
+			pod.Namespace, pod.Name, pod.Spec.Containers[0].Name)
+		deletePods(f, []string{pod.Name})
+		waitForContainerRemoval(pod.Spec.Containers[0].Name, pod.Name, pod.Namespace)
+	}
+}
+
+func runTopologyManagerNegativeTest(f *framework.Framework, numaNodes, numPods int, cpuAmount, sriovResourceName, deviceAmount string) {
+	ginkgo.By(fmt.Sprintf("allocate aligned resources for a %d pod(s): cpuAmount=%s %s=%s", numPods, cpuAmount, sriovResourceName, deviceAmount))
+
+	ctnAttrs := []tmCtnAttribute{
+		{
+			ctnName:       "gu-container",
+			cpuRequest:    cpuAmount,
+			cpuLimit:      cpuAmount,
+			deviceName:    sriovResourceName,
+			deviceRequest: deviceAmount,
+			deviceLimit:   deviceAmount,
+		},
+	}
+
+	podName := "gu-pod"
+	framework.Logf("creating pod %s attrs %v", podName, ctnAttrs)
+	pod := makeTopologyManagerTestPod(podName, numalignCmd, ctnAttrs)
+
+	pod = f.PodClient().Create(pod)
+	err := e2epod.WaitForPodCondition(f.ClientSet, f.Namespace.Name, pod.Name, "Failed", 30*time.Second, func(pod *v1.Pod) (bool, error) {
+		if pod.Status.Phase != v1.PodPending {
+			return true, nil
+		}
+		return false, nil
+	})
+	framework.ExpectNoError(err)
+	pod, err = f.PodClient().Get(pod.Name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	if pod.Status.Phase != v1.PodFailed {
+		framework.Failf("pod %s not failed: %v", pod.Name, pod.Status)
+	}
+	if pod.Status.Reason != topologyError {
+		framework.Failf("pod %s failed for wrong reason: %v", pod.Name, pod.Status)
+	}
+
+	deletePods(f, []string{pod.Name})
+}
+
+func getSRIOVDevicePluginConfigMap(cmFile string) *v1.ConfigMap {
+	cmData := testfiles.ReadOrDie(SRIOVDevicePluginCMYAML)
+	var err error
+
+	// the SRIOVDP configuration is hw-dependent, so we allow per-test-host customization.
+	framework.Logf("host-local SRIOV Device Plugin Config Map %q", cmFile)
+	if cmFile != "" {
+		cmData, err = ioutil.ReadFile(cmFile)
+		if err != nil {
+			framework.Failf("unable to load the SRIOV Device Plugin ConfigMap: %v", err)
+		}
+	} else {
+		framework.Logf("Using built-in SRIOV Device Plugin Config Map")
+	}
+
+	return readConfigMapV1OrDie(cmData)
+}
+
+func runTopologyManagerNodeAlignmentSuiteTests(f *framework.Framework, configMap *v1.ConfigMap, reservedSystemCPUs string, numaNodes, coreCount int) {
+	var err error
+
+	ginkgo.By(fmt.Sprintf("Creating configMap %v/%v", metav1.NamespaceSystem, configMap.Name))
```

Comment:
Perhaps consider moving the SRIOV setup to separate function (setupSRIOVConf()... or something like that). 

Eventually, we will have support for other devices (memory manager, GPU). It seems a little cleaner to have the device setup separately from the actual tests.

Reply:
Done